test/release/examples/primers/specialMethods.chpl

```diff
@@ -0,0 +1,172 @@
+// Special Methods
+
+/* This primer covers special methods for classes and records */
+
+// Classes and records can have a handful of methods that are specially named
+// and treated differently than other methods.  These methods enable
+// operations such as initializing or deinitializing a class or record
+// instance, accessing a class or record as though it were an array,
+// iterating over an class or record, and defining the way a class or record
+// is read from or written to a channel.
+//
+// Note that there are two ways to declare a method. Methods declared within
+// a class or record are called primary methods:
+
+record ExampleRecord1 {
+  var exampleField: int;
+  proc primaryMethod() { }
+}
+
+
+// Methods declared outside of a class or record are called secondary methods:
+
+record ExampleRecord2 {
+  var exampleField: int;
+}
+proc ExampleRecord2.secondaryMethod() { }
+
+// This primer will use the secondary method form, but all of the special
+// methods can also be written as primary methods.
+
+// First we will declare a simple record with a field that is a tuple of
+// integers.  We'll add special methods and iterators to this record later.
+record R {
+  param size: int = 10;
+  var vals: size*int;
+}
+
+/*
+  Initializers and Deinitializers
+  -------------------------------
+*/
+
+// An initializer is called when creating an instance of the class or record,
+// for example with the ``new`` keyword. An initializer accepting zero
+// arguments is called a *default initializer*.  An initializer that takes
+// another instance of the same class or record is called a *copy initializer*.
+
+// The ``deinit`` method will deinitialize a record when it leaves scope,
```

Comment:
It seems like `postinit()` should probably be mentioned here as well (even if its detailed description is left until elsewhere) just since it's another related special method name.